File: c/cap3/cap3.part1ex01.c
/* Copyright (C) 2022-2023 Francisco de Assis Zampirolli - All Rights Reserved
 * Author: Francisco de Assis Zampirolli
 * Purpose: Examples and some exercises from the Information Processing book:
 * https://editora.ufabc.edu.br/matematica-e-ciencias-da-computacao/58-processando-a-informacao
 * Language: C
 */

#include <stdio.h>
#include <math.h>

 // MÉTODOS
float delta(float a, float b, float c) {
  float d = b * b - 4 * a * c;
  return d;
}

float leia() {
  float valor;
  printf("Entre com um valor: ");
  scanf("%f", &valor);
  return valor;
}

// PROGRAMA PRINCIPAL
int main(void) {

  // ENTRADAS
  float a, b, c;
  a = leia();
  b = leia();
  c = leia();

  // PROCESSAMENTO e SAÍDA
  double d = (double)delta(a, b, c);
  printf("Delta = %.1f\n", d);
  if (d < 0) {
    printf("A equacao nao possui rai4zes reais\n");
  }
  if (d == 0) {
    printf("Raiz 1: %.1f\n", (-b + sqrt(d) / 2 * a));
  }
  if (d > 0) {
    printf("Raiz 2: %.1f e %.1f\n", (-b - sqrt(d) / 2 * a), (-b + sqrt(d) / 2 * a));
  }
  return 0;
}

File: py/cap3/cap3.part1ex01.py
"""
Copyright(C) 2022 - 2023 Francisco de Assis Zampirolli - All Rights Reserved
Purpose : Examples and some exercises from the Information Processing book
Language : Python
"""


# DEFINIÇÕES DOS MÉTODOS
def delta(a, b, c):
    return b * b - 4 * a * c


# ENTRADA DE DADOS
print("Calcula as raízes de equação de 2º grau na forma ax^2 + bx + c")
a = float(input("Entre com o primeiro termo a: "))
b = float(input("Entre com o segundo  termo b: "))
c = float(input("Entre com o terceiro termo c: "))

# PROCESSAMENTO E SAÍDA
d = delta(a, b, c)
print("O delta é ", d)
if d < 0:
    print("A equação não possui raízes reais.")
if d == 0:
    print("A raíz é ", -b + d ** (1 / 2.0) / 2 * a)
if d > 0:
    print("Raízes:", -b - d ** (1 / 2.0) / 2 * a, ",", -b + d ** (1 / 2.0) / 2 * a)
